sdks/common/runner-api/src/main/proto/beam_runner_api.proto

```diff
@@ -0,0 +1,436 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Protocol Buffers describing the Runner API, which is the runner-independent,
+ * SDK-independent definition of the Beam model.
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.runner_api.v1;
+
+option java_package = "org.apache.beam.sdks.common.runner_api.v1";
+option java_outer_classname = "RunnerApi";
+
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+
+// A Pipeline contains the entire hierarchical graph of transforms as well
+// as descriptors for all PCollections.
+//
+// The hierarchical graph itself is an independent layer from the payloads
+// specific to the Runner API, for ease of reuse by e.g. the Fn API.
+message Pipeline {
+
+  // (Required) A map from pipeline-scoped id to graph node.
+  //
+  // Each node is required to contain a PTransform specification.
+  map<int64, GraphNode> transform_nodes = 1;
+
+  // (Required) A map from pipeline-scoped id to PCollection.
+  map<int64, PCollection> pcollections = 2;
```

Comment:
I had imagined them being unique throughout the pipeline, though it is only necessary for them to be unique within this map. I like the readability of `pipeline.getPCollections().get(<id>)` more than `pipeline.getThingies().get(<id>).getPCollection()`